wp-content/themes/core/components/blocks/stats/Stats_Block_Controller.php

```diff
@@ -0,0 +1,179 @@
+
+	/**
+	 * @var Statistic_Model[] The collection of stats to render. Each item should be a \Tribe\Project\Templates\Models\Statistic object.
+	 */
+	private array $stats;
+
+	public string $layout;
+	public string $content_align;
+	public string $display_dividers;
+	public string $title;
+	public string $description;
+	public array $container_classes;
+	public array $content_classes;
+	public array $classes;
+	public array $attrs;
+
+	public function __construct( array $args = [] ) {
+		$args = $this->parse_args( $args );
+
+		$this->layout            = (string) $args[ self::LAYOUT ];
+		$this->content_align     = (string) $args[ self::CONTENT_ALIGN ];
+		$this->display_dividers  = (string) $args[ self::DISPLAY_DIVIDERS ];
+		$this->title             = (string) $args[ self::TITLE ];
+		$this->description       = (string) $args[ self::DESCRIPTION ];
+		$this->container_classes = (array) $args[ self::CONTAINER_CLASSES ];
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->stats             = (array) $args[ self::STATS ];
+
+		$this->get_stats();
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function defaults(): array {
+		return [
+			self::LAYOUT            => Stats_Block::LAYOUT_STACKED,
+			self::CONTENT_ALIGN     => Stats_Block::CONTENT_ALIGN_LEFT,
+			self::DISPLAY_DIVIDERS  => Stats_Block::DISPLAY_DIVIDERS,
+			self::TITLE             => '',
+			self::DESCRIPTION       => '',
+			self::CONTAINER_CLASSES => [ 'b-stats__container', 'l-container' ],
+			self::CONTENT_CLASSES   => [ 'b-stats__content' ],
+			self::CLASSES           => [ 'c-block', 'b-stats' ],
+			self::ATTRS             => [],
+			self::STATS             => [],
+		];
+	}
+
+	/**
+	 * @param array $args
+	 *
+	 * @return array
+	 */
+	public function get_content_args(): array {
+		return [
+			'tag'     => 'header',
+			'classes' => [ 'b-stats__header' ],
+			'title'   => defer_template_part( 'components/text/text', null, [
+				'content' => $this->title,
+				'tag'     => 'h2',
+				'classes' => [ 'b-stats__title', 'h3' ],
+			] ),
+			'content' => defer_template_part( 'components/text/text', null, [
+				'content' => $this->description,
+				'classes' => [ 'b-stats__description', 't-sink', 's-sink' ],
+			] ),
+			'layout'  => Stats_Block::CONTENT_ALIGN_CENTER === $this->content_align ? Content_Block::LAYOUT_CENTER : Content_Block::LAYOUT_STACKED,
+		];
+	}
+
+	/**
+	 * Loop through the stats provided setup the statistic components arguments for each.
+	 *
+	 * @return array
+	 */
+	public function get_stats(): array {
+		$statistic_args = [];
+
+		foreach ( $this->stats as $item ) {
+			// Skip over statistic rows with no value.
+			// TODO: fix fatal error caused by calling $item[ Stats_Block::ROW_VALUE ].
+//			if ( empty( $item[ Stats_Block::ROW_VALUE ] ) ) {
+//				continue;
+//			}
```

Comment:
@msawicki I'm trying to skip the entire row if there's no value set for the repeater's ROW_VALUE field but it's causing a fatal when I'm checking the condition like this (http://p.tri.be/V4zQ2C) is there a better way to do this?

Reply:
Imho, I think we should skip this check and just render the empty stat. An editor would have to intentionally _add_ a stat/row without adding any values to it. Maybe they want it blank for a reason. And, if they don't then they can easily just remove the row in the editor. (Just my 2-cents.)